tests/data/rooms.helper.js

```diff
@@ -24,6 +24,28 @@ export const createRoom = ({ name, type, username, members = [] }) => {
 		});
 };
 
+export const deleteRoom = ({ type, roomId }) => {
```

Comment:
what do you think of this? It felt a little repetitive given it is practically identical to `closeRoom`. Could maybe simplify with:
```
function actionRoom({ action, type, roomId }) {
	if (!type) {
		throw new Error(`"type" is required in "${action}Room" test helper`);
	}
	if (!roomId) {
		throw new Error(`"roomId" is required in "${action}Room" test helper`);
	}
	const endpoints = {
		c: 'channels',
		p: 'groups',
		d: 'im',
	};
	return new Promise((resolve) => {
		request.post(api(`${endpoints[type]}.${action}`))
			.set(credentials)
			.send({
				roomId,
			})
			.end(resolve);
	});
};

export const deleteRoom = ({ type, roomId}) => {
	return actionRoom({ action: 'delete', type, roomId });
};

export const closeRoom = ({ type, roomId }) => {
	return actionRoom({ action: 'close', type, roomId });
};
```

Reply:
I like it =)

Reply:
Done. 

Any preference over my comment RE the `im.delete` route and modifying `findRoomByIdOrName` vs a little duplication calling to `getDirectMessageByNameOrIdWithOptionToJoin` in the delete handler? See https://github.com/RocketChat/Rocket.Chat/pull/18022#discussion_r690879734